jfr-streaming/src/main/java/io/opentelemetry/contrib/jfr/metrics/internal/cpu/OverallCPULoadHandler.java

```diff
@@ -50,16 +56,35 @@ public void initializeMeter(Meter meter) {
             measurement -> {
               measurement.record(machine);
             });
+    meter
+        .gaugeBuilder(METRIC_NAME_MACHINE_MINUTE)
+        .setDescription(METRIC_DESCRIPTION_MACHINE_MINUTE)
+        .setUnit(UNIT_UTILIZATION)
+        .buildWithCallback(
+            measurement -> {
+              measurement.record(machineMinuteAverage);
+            });
   }
 
   @Override
   public void accept(RecordedEvent ev) {
+    // Synchronized to avoid races on machineSum
+    synchronized (this) {
+      if (count % 60 == 0) {
+        machineMinuteAverage = machineSum / 60;
+        count = 0;
+        machineSum = 0;
+      }
```

Comment:
I think the most correct thing to do here would be to retain a rolling list of the last 60 measurements. Each time a new measurement comes in, drop the oldest. Then when the callback is invoked take the average of all the entries.

Reply:
Ok it's been changed now. I agree that the suggested approach is better. That way the granularity will be per second instead of per minute.